tests/string_span_tests.cpp

```diff
@@ -1,1217 +1,1218 @@
-}
-
-TEST(string_span_tests, char16_t_type)
-{
-    gsl::cu16string_span<> ss1 = gsl::ensure_z(u"abc");
-    EXPECT_TRUE(ss1.size() == 3);
-    EXPECT_TRUE(ss1.size_bytes() == 6);
-
-    std::u16string s1 = gsl::to_string(ss1);
-    EXPECT_TRUE(s1 == u"abc");
-
-    std::u16string s2 = u"abc";
-    gsl::u16string_span<> ss2 = s2;
-    EXPECT_TRUE(ss2.size() == 3);
-
-    gsl::u16string_span<> ss3 = ss2.subspan(1, 1);
-    EXPECT_TRUE(ss3.size() == 1);
-    EXPECT_TRUE(ss3[0] == u'b');
-
-    char16_t buf[4]{u'a', u'b', u'c', u'\0'};
-    gsl::u16string_span<> ss4{buf, 4};
-    EXPECT_TRUE(ss4[3] == u'\0');
-
-    gsl::cu16zstring_span<> ss5(u"abc");
-    EXPECT_TRUE((ss5.as_string_span().size()) == 3);
-
-    gsl::cu16string_span<> ss6 = ss5.as_string_span();
-    EXPECT_TRUE(ss6 == ss1);
-
-    std::vector<char16_t> v7 = {u'a', u'b', u'c'};
-    gsl::cu16string_span<> ss7{v7};
-    EXPECT_TRUE(ss7 == ss1);
-
-    gsl::cu16string_span<> ss8 = gsl::ensure_z(u"abc");
-    gsl::cu16string_span<> ss9 = gsl::ensure_z(u"abc");
-    EXPECT_TRUE(ss8 == ss9);
-
-    ss9 = gsl::ensure_z(u"abd");
-    EXPECT_TRUE(ss8 < ss9);
-    EXPECT_TRUE(ss8 <= ss9);
-    EXPECT_TRUE(ss8 != ss9);
-}
-
-TEST(string_span_tests, char32_t_type)
-{
-    gsl::cu32string_span<> ss1 = gsl::ensure_z(U"abc");
-    EXPECT_TRUE(ss1.size() == 3);
-    EXPECT_TRUE(ss1.size_bytes() == 12);
-
-    std::u32string s1 = gsl::to_string(ss1);
-    EXPECT_TRUE(s1 == U"abc");
-
-    std::u32string s2 = U"abc";
-    gsl::u32string_span<> ss2 = s2;
-    EXPECT_TRUE(ss2.size() == 3);
-
-    gsl::u32string_span<> ss3 = ss2.subspan(1, 1);
-    EXPECT_TRUE(ss3.size() == 1);
-    EXPECT_TRUE(ss3[0] == U'b');
-
-    char32_t buf[4]{U'a', U'b', U'c', U'\0'};
-    gsl::u32string_span<> ss4{buf, 4};
-    EXPECT_TRUE(ss4[3] == u'\0');
-
-    gsl::cu32zstring_span<> ss5(U"abc");
-    EXPECT_TRUE(ss5.as_string_span().size() == 3);
-
-    gsl::cu32string_span<> ss6 = ss5.as_string_span();
-    EXPECT_TRUE(ss6 == ss1);
-
-    gsl::cu32string_span<> ss8 = gsl::ensure_z(U"abc");
-    gsl::cu32string_span<> ss9 = gsl::ensure_z(U"abc");
-    EXPECT_TRUE(ss8 == ss9);
-
-    ss9 = gsl::ensure_z(U"abd");
-    EXPECT_TRUE(ss8 < ss9);
-    EXPECT_TRUE(ss8 <= ss9);
-    EXPECT_TRUE(ss8 != ss9);
-}
-
-TEST(string_span_tests, as_bytes)
-{
-    cwzstring_span<> v(L"qwerty");
-    const auto s = v.as_string_span();
-    const auto bs = as_bytes(s);
-    EXPECT_TRUE(static_cast<const void*>(bs.data()) == static_cast<const void*>(s.data()));
-    EXPECT_TRUE(bs.size() == s.size_bytes());
-}
-
-TEST(string_span_tests, as_writable_bytes)
-{
-    wchar_t buf[]{L"qwerty"};
-    wzstring_span<> v(buf);
-    const auto s = v.as_string_span();
-    const auto bs = as_writable_bytes(s);
-    EXPECT_TRUE(static_cast<const void*>(bs.data()) == static_cast<const void*>(s.data()));
-    EXPECT_TRUE(bs.size() == s.size_bytes());
-}
+///////////////////////////////////////////////////////////////////////////////

```

Comment:
I suspect similar unnecessary changes here, but I'm not going to audit all 1200 lines.

Reply:
yeah this file is weird. I'm guessing there's an issue with the whitespaces. I'll revert this and adjust everything as necessary

Reply:
There's a 99.99999% chance that you managed to change the line endings from CRLF to LF or vice versa.

Reply:
I think I opened it in VS instead of VSCode and the line endings got borked.

Reply:
Either way, you're right. The file got switched to CRLF for some reason.